### PR TITLE
fix: logout

### PR DESCRIPTION
**Pull request by meniRoy**
fix logout and re-login flow.

- when user log out: we stop environments pulling, clear environment view, clear credentials, show logout message in environments view.
- when logged in user execute login command show "You are already logged in." notification message
- authentication integration test

The credentials used in the video deleted...

https://github.com/env0/vscode-env0/assets/33559192/1730a80d-0d2a-47cb-8938-7eb30ecb06de

### CODE REVIEW

**Comment by meniRoy** [comment]
[Log-out command doesn't work](https://trello.com/c/3HGSDYnt/10558-log-out-command-doesnt-work)

File: src/auth.ts [hunk 1]
```diff
@@ -12,10 +12,16 @@ const env0SecretKey = "env0.secret";
 export class AuthService {
   public onAuth?: () => unknown = undefined;
   constructor(private readonly context: vscode.ExtensionContext) {}
-  public registerLoginCommand() {
+  public registerLoginCommand(onLogin: () => unknown) {
```
Comment (HeverFarber) [inline src/auth.ts:15]:
```suggestion
  public registerLoginCommand(onLogin: () => void) {
```

File: src/auth.ts [hunk 2]
```diff
@@ -46,18 +52,18 @@ export class AuthService {
...
-          this.onAuth = undefined;
+          await onLogin();
         }
       }
     );
     this.context.subscriptions.push(disposable);
   }
 
-  public registerLogoutCommand() {
+  public registerLogoutCommand(onLogOut: () => unknown) {
```
Comment (HeverFarber) [inline src/auth.ts:62]:
```suggestion
  public registerLogoutCommand(onLogout: () => void) {
```

**Review by HeverFarber — APPROVED** [review]

File: src/auth.ts (hunk 1 above)
Comment (meniRoy) [reply src/auth.ts]: c4d88edc6d7f79d26b6ac11810d984aefba4d7d0

File: src/auth.ts (hunk 2 above)
Comment (meniRoy) [reply src/auth.ts]: c4d88edc6d7f79d26b6ac11810d984aefba4d7d0

**Comment by meniRoy** [comment]
[Integration Tests](https://trello.com/c/eoV0MdoJ/10053-integration-tests)

**Pull request merged**